prow.sh

```diff
@@ -0,0 +1,667 @@
+#
+#   When testing an update for a component in a PR job, the
+#   CSI_PROW_DEPLOYMENT variable can be set in the
+#   .prow.sh of each component when there are breaking changes
+#   that require using a non-default deployment. Otherwise
+#   a deployment named "kubernetes-x.yy" is used.
+#   "none" disables the deployment of the hostpath driver.
+#
+# When no deploy script is found (nothing in `deploy` directory,
+# CSI_PROW_HOSTPATH_REPO=none), nothing gets deployed.
+configvar CSI_PROW_HOSTPATH_VERSION master "hostpath driver"
+configvar CSI_PROW_HOSTPATH_REPO https://github.com/kubernetes-csi/csi-driver-host-path "hostpath repo"
+# Ignore 'See if you can use ${variable//search/replace} instead.'
+# shellcheck disable=SC2001
+configvar CSI_PROW_DEPLOYMENT "kubernetes-$(echo "${CSI_PROW_KUBERNETES_VERSION}" | sed -e 's/\([0-9]*\)\.\([0-9]*\).*/\1.\2/')" "deployment"
+
+# If CSI_PROW_HOSTPATH_CANARY is set (typically to "canary", but also
+# "1.0-canary"), then all image versions are replaced with that
+# version tag.
+configvar CSI_PROW_HOSTPATH_CANARY "" "hostpath image"
+
+# The E2E testing can come from an arbitrary repo. The expectation is that
+# the repo supports "go test ./test/e2e -args --storage.testdriver" (https://github.com/kubernetes/kubernetes/pull/72836)
+# after setting KUBECONFIG. As a special case, if the repository is Kubernetes,
+# then `make WHAT=test/e2e/e2e.test` is called first to ensure that
+# all generated files are present.
+#
+# CSI_PROW_E2E_REPO=none disables E2E testing.
+#
+# TODO (?): support also an E2E test suite in the current repository.
+# TODO: upstream Kubernetes
+configvar CSI_PROW_E2E_REPO https://github.com/pohly/kubernetes "E2E repo"
+configvar CSI_PROW_E2E_VERSION storage-external-snapshot "E2E version"
+configvar CSI_PROW_E2E_IMPORT_PATH k8s.io/kubernetes "E2E package"
+
+# These control which tests run resp. don't run under "ginkgo -p".
+# Tests that depend on alpha features are disabled by default because
+# all tests with a feature tag are skipped.
+#
+# Note that feature tags in the test are set based on what is an alpha
+# feature in the *current* Kubernetes. When updating the test via
+# the CSI_PROW_E2E variables above and testing against an older Kubernetes
+# release, some tests might be enabled by default which don't pass
+# for that older Kubernetes version. In that case, the CSI_PROW_E2E_PARALLEL_SKIP_<k8s-version>
+# variable must be extended. <k8s-version> is just major+minor version
+# separated by underscore.
+configvar CSI_PROW_E2E_PARALLEL_FOCUS 'External.Storage.*csi-hostpath' "parallel E2E focus"
+# Raw block was an alpha feature in 1.13.
+configvar CSI_PROW_E2E_PARALLEL_SKIP_1_13 '|\[Testpattern: Dynamic PV .block volmode.\]' "additional parallel test skip for Kubernetes 1.13"
+configvar CSI_PROW_E2E_PARALLEL_SKIP "\\[Feature:|\\[Disruptive\\]$(eval echo "\${CSI_PROW_E2E_PARALLEL_SKIP_${csi_prow_kubernetes_version_suffix}}")" "parallel E2E skip"
```

Comment:
No [Serial]?

Reply:
Ahem, good catch. Will fix.

Reply:
I've introduced the distinction between serial/parallel and stable/alpha and made it possible to run any of these combinations in a job (from one per job to all four).